src/finite_differencing.jl

```diff
@@ -24,14 +33,18 @@ result to back-propagate the sensitivity ȳ. If ȳ, x and v are column vectors
 equivalent to computing `ȳ.'(J f)(x) v`, where `(J f)(x)` denotes the Jacobian of `f`
 evaluated at `x`. Analogous operations happen for scalars and N-dimensional arrays.
 """
-function compute_Dv(f::Function, ȳ::ArrayOr∇Real, x::ArrayOr∇Real, v::ArrayOr∇Real)
-    return sum(∇(f(Leaf(Tape(), x)), ȳ)[1] .* v)
-end
-function compute_Dv(f::Function, ȳ::ArrayOr∇Real, x::∇RealArray, v::∇RealArray)
-    length(x) == length(v) || throw(ArgumentError("length(x) != length(v)."))
-    type_tuple = Tuple{Function, ArrayOr∇Real, ArrayOr∇Real, ArrayOr∇Real}
-    return invoke(approximate_Dv, type_tuple, f, ȳ, x, v)
+function compute_Dv(
+    f::Function,
+    ȳ::ArrayOr∇Real,
+    x::Tuple{Vararg{ArrayOr∇Real}},
+    v::Tuple{Vararg{ArrayOr∇Real}},
+)
+    x_ = Leaf.(Tape(), x)
+    ∇f = ∇(f(x_...), ȳ)
+    return sum(map((x, v)->sum(∇f[x] .* v), x_, v))
 end
+compute_Dv(f::Function, ȳ::ArrayOr∇Real, x::ArrayOr∇Real, v::ArrayOr∇Real) =
+    compute_Dv(f, ȳ, (x,), (v,))
 
 # Compute the absolute and relative errors between x and y respectively.
 compute_errs(x, y) = (abs.(x - y), abs.(x - y) ./ (abs.(x) + 1e-12))
```

Comment:
Since the roundoff error of `x - y` is approximately `eps() * (abs(x) + abs(y))`, it might make more sense to compute the relative error as follows:

```julia
compute_errs(x, y) = (abs.(x - y), abs.(x - y) ./ (abs.(x) + abs.(y)))
```

(I'm not sure if adding a small number to the denominator is necessary as `1/0` simply yields `Inf`.) The above is furthermore consistent with the way `isapprox` computes the relative error.